pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -137,6 +137,10 @@ def advance(self, batch, batch_idx):
             if result:
                 self.batch_outputs[0].append(deepcopy(result.result_collection))
 
+            # clear reference to this step's training loss so that it can be garbage
+            # collected before the next training step
+            trainer._results.minimize = None
```

Comment:
shouldn't it go inside the `automatic_optimization` block right above?

Reply:
Yes, of course